### PR TITLE
Refine test failure category

### PR DESCRIPTION
**Pull request by ChadNedzlek**
The "Test" category includes a lot of build authoring and infrastructure failures.
Let's classify them a bit more granularly.

### CODE REVIEW

**Comment by ChadNedzlek** [comment]
@alexperovich can you check that these categories make sense where I added them

@chcosta does this look reasonable?

**Comment by markwilkie** [comment]
The existence of this PR makes me happy on the inside

File: src/Microsoft.DotNet.Helix/Sdk/FailureCategory.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace Microsoft.DotNet.Helix
+{
+    public struct FailureCategory
+    {
+        public string Value { get; }
+
+        public static FailureCategory Build { get; } = new FailureCategory("Build");
+        public static FailureCategory Test { get; } = new FailureCategory("Test");
+        public static FailureCategory Infrastructure { get; } = new FailureCategory("Infrastructure");
```
Comment (chcosta) [inline src/Microsoft.DotNet.Helix/Sdk/FailureCategory.cs]: `Infrastructure` seems a bit too broad.  If I'm just looking at a dashboard of bucketed failures, "infrastructure" could be just about anything.  Maybe `HelixInfrastructure` or something slightly more specific that would let us know that Helix needs some attention?

Comment (chcosta) [reply src/Microsoft.DotNet.Helix/Sdk/FailureCategory.cs]: Yeah, it's a very hazy area.  We (intentionally to this point), haven't been strict about categorizations.  There's some uncertainty around how much we want to slice things.  Mostly we want to get a big picture look at what is failing.  I think "Infrastructure" is too big of a bucket.

The current goal of the categories is to identify key pieces that we should invest in to improve.  ie, We don't necessarily need to know that an issue was a Helix bug or network reliability or a machine thing, but we need to know that Helix is a pain point and we need to look closer / invest in Helix to improve things.  

I think it's ok to have a bunch of categories nobody cares about.  When they become problematic and start popping up more, people will start caring.

Comment (ChadNedzlek) [reply src/Microsoft.DotNet.Helix/Sdk/FailureCategory.cs]: Done! Infrastructure -> Helix

**Pull request merged**